d2/src/main/java/com/linkedin/d2/xds/LoadBalanceTaskPool/NewLoadBalancerTaskthreadPool.java

```diff
@@ -0,0 +1,29 @@
+package com.linkedin.d2.xds.LoadBalanceTaskPool;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+
+public class NewLoadBalancerTaskthreadPool {
+  private ThreadPoolExecutor _threadPoolExecutor;
+  private int _corePoolSize = 2;
+  private int _maximumPoolSize = 3;
+  private long _keepAliveTime = 200;
+  private int _queueSize = 1000;
+
+  public NewLoadBalancerTaskthreadPool() {
+    RejectedExecutionHandler rejectedExecutionHandler = new NewBalanceTaskRejectedPolicy();
+    _threadPoolExecutor = new ThreadPoolExecutor(_corePoolSize, _maximumPoolSize, _keepAliveTime, TimeUnit.MILLISECONDS,
+        new ArrayBlockingQueue<Runnable>(_queueSize), rejectedExecutionHandler);
+  }
+
+  public void execute(Runnable task) {
+    _threadPoolExecutor.execute(task);
+  }
+
+  public void shutdown() {
+    _threadPoolExecutor.shutdown();
+  }
+}
```

Comment:
I don't think its' a good idea to have a custom wrapper for what's just a normal threadpool. This is especially true in the context of LinkedIn code where we have to use threadpools provided by Offspring for monitoring purposes. You can delete this class and inject a threadpool in DualReadZkAndXdsLoadBalancerFactory instead.